MathLib/LinAlg/Eigen/EigenMatrix.h

```diff
@@ -170,18 +168,17 @@ class EigenMatrix final
     RawMatrixType _mat;
 };
 
-
-template<class T_DENSE_MATRIX>
-void
-EigenMatrix::add(std::vector<std::size_t> const& row_pos, std::vector<std::size_t> const& col_pos,
-        const T_DENSE_MATRIX &sub_matrix, double fkt)
+template <class T_DENSE_MATRIX>
+void EigenMatrix::add(std::vector<IndexType> const& row_pos,
+                      std::vector<IndexType> const& col_pos,
+                      const T_DENSE_MATRIX& sub_matrix, double fkt)
 {
-    const std::size_t n_rows = row_pos.size();
-    const std::size_t n_cols = col_pos.size();
-    for (std::size_t i = 0; i < n_rows; i++) {
-        const std::size_t row = row_pos[i];
-        for (std::size_t j = 0; j < n_cols; j++) {
-            const std::size_t col = col_pos[j];
+    const IndexType n_rows = row_pos.size();
+    const IndexType n_cols = col_pos.size();
```

Comment:
`row_pos` and `col_pos` are of type `std::vector`. The `size()` method returns `size_type` which has not to be equal to `IndexType`. So maybe these changes are not necessary?